05_stringer/harirakr/main_test.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	assert.Equal(t, "127.0.0.1\n", buf.String())
+
+}
+
+func TestStringer(t *testing.T) {
+	assert.Equal(t, "0.0.0.0", fmt.Sprintf("%v", IPAddr{}))
```

Comment:
You should look at structuring these as a table of tests, since they all follow the same structure (take an input, get a response and assert for equality.

Reply:
Done.